src/main/java/nus/edu/rizzler/Rizzler.java

```diff
@@ -0,0 +1,97 @@
+                     to add a EVENT nus.edu.rizzler.task type 'event [nus.edu.rizzler.task] FROM [start time] TO [end time]' , and nus.edu.rizzler.Rizzler's got it handled!\s
+                  2. Need the full rundown? Type 'list' and I’ll drop the goods!
+                  3. Crushed a nus.edu.rizzler.task? Type 'mark [nus.edu.rizzler.task number]' and let's celebrate!
+                  4. Changed your mind? Type 'undo [nus.edu.rizzler.task number]' and I'll reset that for you!
+                  5. Wanna bounce? Just hit me with 'bye' and I'll catch you on the flip side!
+                  --------------------------------------------------------
+                 \s""";
+
+        System.out.println(MENU);
+    }
+
+    public void listenCommand() {
+        Scanner scanner = new Scanner(System.in);
+        String command = scanner.nextLine();
+
+        while (!command.equals("bye")) {
+            try {
+                handleCommand(command);
+            } catch (InvalidInputException e) {
+                System.out.println(e.getMessage());
+            }
+            command = scanner.nextLine();
+        }
+        System.out.println("Peace out, Rizzler’s got places to be! " + emoji.getCoolFaceEmoji() + emoji.getFistBumpEmoji());
+    }
+
+    private void handleCommand(String command) throws InvalidInputException {
+        TaskManager taskManager = new TaskManager();
+
+        if (command.equals("list")) {
+            taskManager.displayTaskList();
+
+        } else if (command.startsWith("mark")) {
+            try {
+                int taskNumber = Integer.parseInt(command.split(" ")[1]);
+                taskManager.completeTask(taskNumber);
+            } catch (NumberFormatException e) {
+                System.out.println("ERROR: Task number must be an integer --> 'mark [task number]'" + emoji.getExclamationMarkEmoji());
+            } catch (IndexOutOfBoundsException e) {
+                System.out.println("ERROR: Please follow the format specified in the MENU --> 'mark [task number]'" + emoji.getExclamationMarkEmoji());
+            }
+
+        } else if (command.startsWith("undo")) {
+            try {
+                int taskNumber = Integer.parseInt(command.split(" ")[1]);
+                taskManager.undoTask(taskNumber);
+            } catch (NumberFormatException e) {
+                System.out.println("ERROR: Task number must be an integer 'undo [task number]'" + emoji.getExclamationMarkEmoji());
+            } catch (IndexOutOfBoundsException e) {
+                System.out.println("ERROR: Please follow 'undo [task number]' format" + emoji.getExclamationMarkEmoji());
+            }
+
+        } else if (command.startsWith("todo") || command.startsWith("deadline") || command.startsWith("event")) {
+            taskManager.addTask(command);
+
+        } else {
+            throw new InvalidInputException("That's an invalid command! " + emoji.getCrossMarkEmoji() + emoji.getCrossMarkEmoji());
+        }
+    }
```

Comment:
This section of code may be prone to nested statements. You can consider improving it to handle each case separately.